operator/bucketcontroller/webhook.go

```diff
@@ -0,0 +1,54 @@
+package bucketcontroller
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	exoscalev1 "github.com/vshn/provider-exoscale/apis/exoscale/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// BucketValidator validates admission requests.
+type BucketValidator struct {
+	log logr.Logger
+}
+
+// ValidateCreate implements admission.CustomValidator.
+func (v *BucketValidator) ValidateCreate(_ context.Context, obj runtime.Object) error {
+	bucket := obj.(*exoscalev1.Bucket)
+	v.log.V(1).Info("Validate create", "name", bucket.Name)
+
+	providerConfigRef := bucket.Spec.ProviderConfigReference
+	if providerConfigRef == nil || providerConfigRef.Name == "" {
+		return fmt.Errorf("a bucket named %q requires a provider config reference",
+			bucket.Name)
+	}
+	return nil
+}
+
+// ValidateUpdate implements admission.CustomValidator.
+func (v *BucketValidator) ValidateUpdate(_ context.Context, oldObj, newObj runtime.Object) error {
+	newBucket := newObj.(*exoscalev1.Bucket)
+	oldBucket := oldObj.(*exoscalev1.Bucket)
+	v.log.V(1).Info("Validate update")
+
+	if oldBucket.Status.AtProvider.BucketName != "" {
```

Comment:
We still need to check if the `spec.providerConfigRef` contains a name (like in ValidateCreate()`. Otherwise, the user can create a resource with ref set, but then remove again in an update operation.